### PR TITLE
[OMNIML-2791] Use nemotron post training dataset for calibration

### PR DESCRIPTION
**Pull request by cjluo-nv**
## What does this PR do?

**Type of change:** ? Update default calibration dataset
**Overview:** ?

We now plan to use the mixture of the `cnn_dailymail` and `nvidia/Nemotron-Post-Training-Dataset-v2` as the default dataset as it shows overall the same or better model accuracy after PTQ, especially for AWQ tasks.

This PR also

1) Add the nvidia/Nemotron-Post-Training-Dataset-v1 dataset which includes the reasoning as an alternative
2) Add --calib_seq flag in the PTQ script so user can specify the max calib sample seq length
3) Increase the max calib batch size from 64 to 512 to accelerate PTQ if the batch size is auto inferred.

## Testing
Benchmarked with gpqa and AIME comparing cnn_dailymail calibration vs nemotron dataset calibration.

## Summary by CodeRabbit

* New Features
  * Default calibration dataset updated to Nemotron Post-Training Dataset v2, with optional v1 support.
  * Added CLI option to set calibration sequence length (calib_seq, default 512).
  * Scripts now accept CALIB_SEQ to pass calibration sequence length through command-line.
  * Expanded calibration batch size handling to support larger batches for improved throughput.
* Documentation
  * Changelog updated to reflect new datasets, default selection, and calib_seq option.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -11,6 +11,8 @@ Model Optimizer Changelog (Linux)
 - Add flag ``op_types_to_exclude_fp16`` in ONNX quantization to exclude ops from being converted to FP16/BF16. Alternatively, for custom TensorRT ops, this can also be done by indicating ``'fp32'`` precision in ``trt_plugins_precision``.
 - Add LoRA mode support for MCore in a new peft submodule: ``modelopt.torch.peft.update_model(model, LORA_CFG)``.
 - Support PTQ and fakequant in vLLM for fast evaluation of arbitrary quantization formats. See ``examples/vllm_serve`` for more details.
+- Add support for ``nemotron-post-training-dataset-v2`` and ``nemotron-post-training-dataset-v1`` in ``examples/llm_ptq``. Default to ``nemotron-post-training-dataset-v2`` if no dataset is specified.
+- Allow specifying ``calib_seq`` in ``examples/llm_ptq`` to set the maximum sequence length for calibration.
```
Comment (sugunav14) [inline CHANGELOG.rst:15]: general question, what is the difference between calib_size and calib_seq?

File: examples/llm_ptq/scripts/parser.sh [hunk 2]
```diff
@@ -64,19 +64,24 @@ parse_options() {
...
       --low_memory_mode ) LOW_MEMORY_MODE=true; shift;;
       --calib_dataset ) CALIB_DATASET="$2"; shift 2;;
+      --calib_seq ) CALIB_SEQ="$2"; shift 2;;
       -- ) shift; break ;;
       * ) break ;;
     esac
   done
 
   DEFAULT_CALIB_SIZE=512
+  DEFAULT_CALIB_SEQ=512
```
Comment (kevalmorabia97) [inline examples/llm_ptq/scripts/parser.sh:74]: Qq. Do we have any study whether PTQ results improve on using larger seq len? For Pruning we generally use 4096/8192 seq len.

Also perhaps `calib_seqlen` would be clearer name than `calib_seq`?

File: CHANGELOG.rst [hunk 3]
```diff
@@ -11,6 +11,8 @@ Model Optimizer Changelog (Linux)
 - Add flag ``op_types_to_exclude_fp16`` in ONNX quantization to exclude ops from being converted to FP16/BF16. Alternatively, for custom TensorRT ops, this can also be done by indicating ``'fp32'`` precision in ``trt_plugins_precision``.
 - Add LoRA mode support for MCore in a new peft submodule: ``modelopt.torch.peft.update_model(model, LORA_CFG)``.
 - Support PTQ and fakequant in vLLM for fast evaluation of arbitrary quantization formats. See ``examples/vllm_serve`` for more details.
+- Add support for ``nemotron-post-training-dataset-v2`` and ``nemotron-post-training-dataset-v1`` in ``examples/llm_ptq``. Default to ``nemotron-post-training-dataset-v2`` if no dataset is specified.
```
Comment (kevalmorabia97) [inline CHANGELOG.rst]:
```suggestion
- Add support for ``nemotron-post-training-dataset-v2`` and ``nemotron-post-training-dataset-v1`` in ``examples/llm_ptq``. Default changed from ``cnn_dailymail`` to ``nemotron-post-training-dataset-v2`` if no dataset is specified.
```

File: examples/llm_ptq/scripts/parser.sh (hunk 2 above)
Comment (cjluo-nv) [reply examples/llm_ptq/scripts/parser.sh:74]: I tested 2048, it does not help

File: CHANGELOG.rst (hunk 1 above)
Comment (cjluo-nv) [reply CHANGELOG.rst:15]: calib_size is the num of calib samples.
calib_seq is the length of the max calib sample sequence.

Comment (realAsma) [reply CHANGELOG.rst:15]: nit:
@cjluo-nv should we use be more verbose and use `--calib_seq_len` for clarity?

Comment (cjluo-nv) [reply CHANGELOG.rst:15]: ACK. Personally I prefer shorter flags.
